ding/framework/tests/test_middleware_pace_controller.py

```diff
@@ -2,45 +2,75 @@
 import unittest
 from unittest import mock
 from unittest.mock import patch
-import pathlib as pl
-import os
-import shutil
+import math
+import time
 from typing import Callable
 
 from ding.framework import Task, Context
 from ding.framework import Parallel
 from ding.framework.middleware import pace_controller
 
 
-def parallel_main():
+def fn(task: "Task"):
 
-    def fn(task: "Task"):
-        another_node_total_step = 0
+    def _fn(ctx: "Context"):
+        time.sleep(0.01)
 
-        def _listen_total_step(total_step):
-            nonlocal another_node_total_step
-            another_node_total_step = total_step
-            return
+    return _fn
 
-        task.on("total_step", _listen_total_step)
 
-        def _fn(ctx: "Context"):
-            nonlocal another_node_total_step
-            assert ctx.total_step <= another_node_total_step + 1
-            assert ctx.total_step >= another_node_total_step
-            task.emit("total_step", ctx.total_step, only_local=True)
-            return
+def parallel_main(theme: str = "", timeout: float = math.inf, if_test_identity: bool = False):
+    with Task(async_mode=True) as task:
+        max_step = 10
+
+        def _listen_to_finish(value):
+            if if_test_identity and task.router.node_id > 0:
+                assert task.ctx.total_step <= max_step
+            else:
+                assert task.ctx.total_step >= max_step - 1
+                assert task.ctx.total_step <= max_step
+
+        task.on("finish", _listen_to_finish)
+
+        identity = str(task.router.node_id)
+        if not if_test_identity:
+            identity = ""
+
+        if task.router.node_id > 0:
+            task.use(fn(task))
+            task.use(pace_controller(task, theme=theme, identity=identity, timeout=timeout))
```

Comment:
Here each process will use a unique identity, is this the expected behavior?

Reply:
yes, I think you are right. I will change the test topology into a mesh mode rather than a star mode.

Reply:
And assign same identity to multiple nodes to test the expected behavior.